services/worker-manager/src/api.js

```diff
@@ -5,6 +5,8 @@ const { ApiError, Provider } = require('./providers/provider');
 const { UNIQUE_VIOLATION } = require('taskcluster-lib-postgres');
 const { WorkerPool, WorkerPoolError, Worker } = require('./data');
 const { createCredentials } = require('./util');
+const { joinTaskQueueId, addSplitFields } = require('../../queue/src/utils');
```

Comment:
I think services should not be aware of other service's internals. Can we move that to a common lib instead?

Reply:
Actually those are not needed at all, worker manager has own utils:

https://github.com/taskcluster/taskcluster/blob/5a25a717299e9eae61d7ed0935ceb028b8319a26/services/worker-manager/src/util.js#L12-L25

Reply:
Perfect, thanks for pointing this out. And the point where services should not be aware of other services' internals. 👍🏻

Reply:
Thanks for spotting this!  100% agree that services shouldn't be referencing each other (and IIRC that's somewhere in the dev docs).

Reply:
Indeed, that's in the best practices doc. https://github.com/taskcluster/taskcluster/blob/main/dev-docs/best-practices/microservices.md#independence